W3F grant evidence submission

xcavate2022 opened the pull request:
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [x] This pull request is being made by the same account as the accepted application.
- [x] I have disclosed any and all sources of reused code in the submitted repositories and have done my due diligence to meet its license requirements.
- [x] In case of acceptance, the payment will be transferred to the BTC/ETH/fiat account provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://grants.web3.foundation/docs/Support%20Docs/milestone-deliverables-guidelines).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/1649

semuelle commented on Xcavate-milestone_1.md:
```diff
@@ -0,0 +1,39 @@
+
+**The [invoice form :pencil:](https://docs.google.com/forms/d/e/1FAIpQLSfmNYaoCgrxyhzgoKQ0ynQvnNRoTmgApz9NrMp-hd8mhIiO0A/viewform) has been filled out correctly for this milestone and the delivery is according to the official [milestone delivery guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/milestone-deliverables-guidelines.md).**  
+
+* **Application Document:** [https://github.com/w3f/Grants-Program/pull/1649]
```
```suggestion
* **Application Document:** https://github.com/w3f/Grants-Program/blob/master/applications/Xcavate.md
```

semuelle reviewed (changes requested):
Thanks for the delivery, @xcavate2022. The pull request should only contain the delivery document, though. Could you move the other document and image elsewhere?

xcavate2022 commented:
apologies document removed - please check now

semuelle commented:
> apologies document removed - please check now

The PR still contains an image. Also, please review the change suggestion above.

xcavate2022 commented:
Apologies - I didn’t realise one of my team had uploaded that image thinking it would help… I’ve deleted it

xcavate2022 commented:
Just checking you are happy with the submission now and not waiting on us?

semuelle commented:
Thanks for the update, @xcavate2022. Looks good so far. Someone is going to review the delivery in the coming days.

dsm-w3f commented:
@xcavate2022 thank you for the milestone delivery. I tried to build the node provided but had some problems with the dependencies. Could you please take a look? Follows the error log of the `cargo build --release`. I tried with two different machines and had the same problem. This prevented me from continuing the evaluation.

```
user@localhost:~/Documents/xcavate/MVP_Lending_Pool$ cargo build --release    
Updating git repository `https://github.com/paritytech/substrate.git`
error: failed to get frame-benchmarking as a dependency of package node-template v4.0.0-dev (/home/user/Documents/xcavate/MVP_Lending_Pool/node)
Caused by:  failed to load source for dependency frame-benchmarking
Caused by:  Unable to update https://github.com/paritytech/substrate.git?branch=polkadot-v1.0.0#98f2e345
Caused by:  object not found - no match for id (98f2e3451c9143278ec53c6718940aeabcd3b68a); class=Odb (9); code=NotFound (-3)
```

xcavate2022 commented:
Thanks for your message DiogoI’ve been told by my team it’s a rust issue… can you please check your version.Default host: x86_64-unknown-linux-gnurustup home:  /home/recrafter/.rustupinstalled toolchains--------------------stable-x86_64-unknown-linux-gnu (default)nightly-2023-01-01-x86_64-unknown-linux-gnunightly-x86_64-unknown-linux-gnuinstalled targets for active toolchain--------------------------------------wasm32-unknown-unknownx86_64-unknown-linux-gnuactive toolchain----------------nightly-2023-01-01-x86_64-unknown-linux-gnu (overridden by '/home/recrafter/substrate/MVP_Lending_Pool/rust-toolchain.toml')rustc 1.68.0-nightly (574b64a97 2022-12-31)Sent from my iPhoneOn 12 Sep 2023, at 02:47, Diogo ***@***.***> wrote:﻿
@xcavate2022 thank you for the milestone delivery. I tried to build the node provided but had some problems with the dependencies. Could you please take a look? Follows the error log of the cargo build --release. I tried with two different machines and had the same problem. This prevented me from continuing the evaluation.
***@***.***:~/Documents/xcavate/MVP_Lending_Pool$ cargo build --release    
Updating git repository `https://github.com/paritytech/substrate.git`
error: failed to get frame-benchmarking as a dependency of package node-template v4.0.0-dev (/home/user/Documents/xcavate/MVP_Lending_Pool/node)
Caused by:  failed to load source for dependency frame-benchmarking
Caused by:  Unable to update https://github.com/paritytech/substrate.git?branch=polkadot-v1.0.0#98f2e345
Caused by:  object not found - no match for id (98f2e3451c9143278ec53c6718940aeabcd3b68a); class=Odb (9); code=NotFound (-3)


—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

xcavate2022 commented:
Sorry it was a dependency issue… if you have the correct version of rust (please see previous message) it should work now.Sent from my iPhoneOn 12 Sep 2023, at 21:29, Diogo ***@***.***> wrote:﻿
@xcavate2022 thanks for the answer. Here is the Rust version that I'm using, however, I think maybe this is not the problem since the Rust toolchain is configured by the project. Could this be caused by the substrate repository having been archived recently and the new version being in another repository?
rustup show
Default host: x86_64-unknown-linux-gnu
rustup home:  /home/user/.rustup

installed toolchains
--------------------

stable-x86_64-unknown-linux-gnu (default)
nightly-2023-01-01-x86_64-unknown-linux-gnu
nightly-x86_64-unknown-linux-gnu
1.69.0-x86_64-unknown-linux-gnu

installed targets for active toolchain
--------------------------------------

wasm32-unknown-unknown
x86_64-unknown-linux-gnu

active toolchain
----------------

nightly-2023-01-01-x86_64-unknown-linux-gnu (overridden by '/home/user/Documents/xcavate/MVP_Lending_Pool/rust-toolchain.toml')
rustc 1.68.0-nightly (574b64a97 2022-12-31)


—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

dsm-w3f commented:
@xcavate2022 thank you for the answer. I did a first round of evaluation. Please see the [evaluation document](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md) and provide proper answers and fixes. Let me know when I can continue this evaluation.

xcavate2022 commented:
Thanks for the comments… my team is working on them. Please find the .env backend details for the DID credential attestation etc```PORT="9090"JWT_SECRET="XCavate JWT Secret"URL=http://localhost:9090BLOCKCHAIN_ENDPOINT=wss://peregrine.kilt.io/parachain-public-wsDID=did:kilt:4skimcqA5SDHsp4K6XM6nQVZSuCPAixbjW6MUok6e5uJqtufSECRET_PAYER_MNEMONIC=forest turn anchor because angry miracle slot unhappy claim blood champion dolphinSECRET_AUTHENTICATION_MNEMONIC=cage tunnel resist radio lab cost quick slight axis mad ankle eraSECRET_ASSERTION_METHOD_MNEMONIC=view science pistol skull enlist bleak wave category lawn real drill balconySECRET_KEY_AGREEMENT_MNEMONIC=curtain chest safe roast place avocado faculty duck dial bless pill mutualADMIN_USERNAME=exampleADMIN_PASSWORD=attester# ***@***.***/xcavate-dev# AWS S3S3_ACCESS_KEY=AKIA3CXDWBJJJHPJBUBJS3_SECRET_KEY=IoevHCuSBy2jmmyIjSAwWcOaAArZEqeGF7K5cdDbAWS_REGION=us-east-1S3_BUCKET=xcavate-cdnCDN_URL=https://xcavate-cdn.s3.amazonaws.com# Crust network adminCRUST_NETWORK_ADMIN_MNEMONIC=shoot coil invest fever output approve follow eight parent soul muffin fewCRUST_NETWORK_ADMIN_ADDRESS=5D7hbnc9oH5Dfhv1txZqyCR2vroHbXGKgk8S2jt1L3jZAieKCRUST_NETWORK_ENDPOINT=wss://rpc-rocky.crust.networkCRUST_IPFS_W3GW=https://crustipfs.xyz```Sent from my iPhoneOn 14 Sep 2023, at 03:45, Diogo ***@***.***> wrote:﻿
@xcavate2022 thank you for the answer. I did a first round of evaluation. Please see the evaluation document and provide proper answers and fixes. Let me know when I can continue this evaluation.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

xcavate2022 commented:
We built an image using this Dockerfile. but it doesn't run with ws so it doesn’t work with the dapps



so it doesn
't interacts with the dapps

# This is an example build stage for the node template. Here we create the binary in a temporary image.

# This is a base image to build substrate nodes
FROM docker.io/paritytech/ci-linux:production as builder

WORKDIR /node-template
COPY . .
RUN rm -rf /usr/local/rustup /usr/local/cargo && \
    curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain none && \
    rustup show && \
    cargo build --release

# This is the 2nd stage: a very small image where we copy the binary."
FROM docker.io/library/ubuntu:22.04
LABEL description="Multistage Docker image for Substrate Node Template" \
  image.type="builder" \
  ***@***.***" \
  image.vendor="Substrate Developer Hub" \
  image.description="Multistage Docker image for Substrate Node Template" \
  image.source="https://github.com/substrate-developer-hub/substrate-node-template" \
  image.documentation="https://github.com/substrate-developer-hub/substrate-node-template"

# Copy the node binary.
COPY --from=builder /node-template/target/release/node-template /usr/local/bin

RUN useradd -m -u 1000 -U -s /bin/sh -d /node-dev node-dev && \
  mkdir -p /chain-data /node-dev/.local/share && \
  chown -R node-dev:node-dev /chain-data && \
  ln -s /chain-data /node-dev/.local/share/node-template && \
  # unclutter and minimize the attack surface
  rm -rf /usr/bin /usr/sbin && \
  # check if executable works in this container
  /usr/local/bin/node-template --version

USER node-dev

EXPOSE 30333 9933 9944 9615
VOLUME ["/chain-data"]

ENTRYPOINT ["/usr/local/bin/node-template"]

> On 13 Sep 2023, at 20:45, Diogo ***@***.***> wrote:
> 
> 
> @xcavate2022 <https://github.com/xcavate2022> thank you for the answer. I did a first round of evaluation. Please see the evaluation document <https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md> and provide proper answers and fixes. Let me know when I can continue this evaluation.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/w3f/Grant-Milestone-Delivery/pull/992#issuecomment-1718219370>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/A2OSX7HAUZLIKJX6SC7GTL3X2IENXANCNFSM6AAAAAA4IZVZMA>.
> You are receiving this because you were mentioned.
>

xcavate2022 commented:
Evaluation V1

 <https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md#pull-request>Pull Request

I noticed the file isn't following the number for the deliverables. For example, the Docker, article and tutorial should be in the "0x.”.

Action: Numbering changed to match

 <https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md#license>License

The license for Xcavate Loan App Repo <https://github.com/XcavateBlockchain/lending_protocol_contracts/blob/main/LICENSE> has some merge conflicts.

Action: Conflicts resolved

 <https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md#xcavate-node>Xcavate Node

I had some problems with this repo. When I tried to use cargo commands, I received this error:

I tried to use a Dockerfile, but I received the same error.

Action: We could not get Dockerfile to work with this node version

I removed the rust-toolchain.toml. It solved the problem for me, but I used the same versions.

Maybe the overridden version by the rust-toolchain.toml doesn't have some caches in comparison to the not overridden versions. Because I tried again with Docker without success.

Action: Xcavate node updated

 <https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md#docker>Docker

Could you explain the problem you faced with the Docker image? You could create a dockerfile for the user to build the image.

Action: See earlier email response with error

 <https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md#pallets>Pallets

I was able to build the loan contract, but when I deployed the contract it didn't appear on the contracts page and the message contracts.instantiateWithCode   ready in the corner didn't disappear. Besides that, I was able to add the existing contract using Polkadot.js because the code was stored and the contract instantiated.

I followed the video tutorial to test the pallets without problems. Watching the tutorial, I didn't notice any usage of the Community Loan Staking pallet. Is it used in the Loan management pallet or somewhere else?

Could you explain the Verification pallet? I saw the comment that is not pallet, this is somewhat confusing.

I tried to run cargo test, but received these errors.

 <https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md#decentralized-developer-loan-dapp>Decentralized Developer Loan dApp

I added the variables in the .env. I noticed the MONGO_DB_URL isn't showing in the example in the readme.

No problems running the repositories.

I disabled the CORS to use the front end. I added the wallet extension and tried to connect my wallet, but it didn't connect, and the console returned this error.

 <https://user-images.githubusercontent.com/112647953/267771009-1751eb01-450d-4fb3-a4b9-20059ccb7e98.png>
Sometimes the backend crashes. I don't know how, but it occurred when I tried to navigate on the page.

Action: Backend updated now

 <https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md#code-quality>Code Quality

I ran cargo clippy and got some warnings in the MVP_Lending_Pool. For example:

I ran in the lending_protocol_contracts too. No warnings. I ran yarn lint in the MVP_Admin and got some problems, for example:


> On 13 Sep 2023, at 20:45, Diogo ***@***.***> wrote:
> 
> 
> @xcavate2022 <https://github.com/xcavate2022> thank you for the answer. I did a first round of evaluation. Please see the evaluation document <https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md> and provide proper answers and fixes. Let me know when I can continue this evaluation.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/w3f/Grant-Milestone-Delivery/pull/992#issuecomment-1718219370>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/A2OSX7HAUZLIKJX6SC7GTL3X2IENXANCNFSM6AAAAAA4IZVZMA>.
> You are receiving this because you were mentioned.
>

xcavate2022 commented:
Action: We have worked on the staking pallet to make it functional. We are able to Stake and Unstake XCAV coins and the staking pallet interacts with the community loan pallet as intended. Stakers can earn staking rewards depending on the loan APYs. Further work is required - to be continued under the next milestone.

> On 14 Sep 2023, at 18:28, Richard Xcav ***@***.***> wrote:
> 
> Evaluation V1
> 
>  <https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md#pull-request>Pull Request
> 
> I noticed the file isn't following the number for the deliverables. For example, the Docker, article and tutorial should be in the "0x.”.
> 
> Action: Numbering changed to match
> 
>  <https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md#license>License
> 
> The license for Xcavate Loan App Repo <https://github.com/XcavateBlockchain/lending_protocol_contracts/blob/main/LICENSE> has some merge conflicts.
> 
> Action: Conflicts resolved
> 
>  <https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md#xcavate-node>Xcavate Node
> 
> I had some problems with this repo. When I tried to use cargo commands, I received this error:
> 
> I tried to use a Dockerfile, but I received the same error.
> 
> Action: We could not get Dockerfile to work with this node version
> 
> I removed the rust-toolchain.toml. It solved the problem for me, but I used the same versions.
> 
> Maybe the overridden version by the rust-toolchain.toml doesn't have some caches in comparison to the not overridden versions. Because I tried again with Docker without success.
> 
> Action: Xcavate node updated
> 
>  <https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md#docker>Docker
> 
> Could you explain the problem you faced with the Docker image? You could create a dockerfile for the user to build the image.
> 
> Action: See earlier email response with error
> 
>  <https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md#pallets>Pallets
> 
> I was able to build the loan contract, but when I deployed the contract it didn't appear on the contracts page and the message contracts.instantiateWithCode   ready in the corner didn't disappear. Besides that, I was able to add the existing contract using Polkadot.js because the code was stored and the contract instantiated.
> 
> I followed the video tutorial to test the pallets without problems. Watching the tutorial, I didn't notice any usage of the Community Loan Staking pallet. Is it used in the Loan management pallet or somewhere else?
> 
> Could you explain the Verification pallet? I saw the comment that is not pallet, this is somewhat confusing.
> 
> I tried to run cargo test, but received these errors.
> 
>  <https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md#decentralized-developer-loan-dapp>Decentralized Developer Loan dApp
> 
> I added the variables in the .env. I noticed the MONGO_DB_URL isn't showing in the example in the readme.
> 
> No problems running the repositories.
> 
> I disabled the CORS to use the front end. I added the wallet extension and tried to connect my wallet, but it didn't connect, and the console returned this error.
> 
>  <https://user-images.githubusercontent.com/112647953/267771009-1751eb01-450d-4fb3-a4b9-20059ccb7e98.png>
> Sometimes the backend crashes. I don't know how, but it occurred when I tried to navigate on the page.
> 
> Action: Backend updated now
> 
>  <https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md#code-quality>Code Quality
> 
> I ran cargo clippy and got some warnings in the MVP_Lending_Pool. For example:
> 
> I ran in the lending_protocol_contracts too. No warnings. I ran yarn lint in the MVP_Admin and got some problems, for example:
> 
> 
>> On 13 Sep 2023, at 20:45, Diogo ***@***.***> wrote:
>> 
>> 
>> @xcavate2022 <https://github.com/xcavate2022> thank you for the answer. I did a first round of evaluation. Please see the evaluation document <https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md> and provide proper answers and fixes. Let me know when I can continue this evaluation.
>> 
>> —
>> Reply to this email directly, view it on GitHub <https://github.com/w3f/Grant-Milestone-Delivery/pull/992#issuecomment-1718219370>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/A2OSX7HAUZLIKJX6SC7GTL3X2IENXANCNFSM6AAAAAA4IZVZMA>.
>> You are receiving this because you were mentioned.
>> 
>

dsm-w3f commented:
@xcavate2022 thank you for the improvements and fixes. I'm still having problems to test the system. Please see the [evaluation document](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md) for details. Does not seem that all problems were fixed. Furthermore, it seems that some deliverables are still in development. Please let me know when all of them are ready to be evaluated again.

meanking commented:
@dsm-w3f, thank you for your feedback.
Just I am curious about the problem you've faced for the wallet connection. `The problem with the Backend is now solved. However, I still can't connect my wallet. `
Could you detail about this, please? So, I can modify something or help you to run it successfully.

RecrafterH commented:
@dsm-w3f hello, for the node we have an updated version on this repo: https://github.com/XcavateBlockchain/Xcavate_Node and for the smart contract the updated repo is this one: https://github.com/XcavateBlockchain/xcavate_loan_contract . Please use these repos for any further evaluation. Regarding the testing of the node and contract please use this polkadot js app site: https://polkadotjs-apps.web.app/#/accounts on the other one there seems to be an issue with uploading smart contracts. Thank you.

xcavate2022 commented:
Hi Diogo,

Many thanks for your most recent response.

Please see actions taken…

1. Wallet Connection Issues: I’m sorry there was the wrong link to the test Sporran Wallet download in the instructions document PDF… should be ok now (same PDF link should work if you empty your browser cache).
2. DAO Pallet: Through our roadmap we will test using a basic voting structure for the Real Estate Developer Loans (which we have incorporated in to the Community Loan Pallet), then we will build out the Governance structure to make the system more decentralised over time (similar to Polkadot and other chains).
New testing video showing voting & staking... https://youtu.be/Bb5ZqYb_LkE <https://youtu.be/Bb5ZqYb_LkE>
3. Verification Pallet: This has been created using Acurast (rather than Chainlink) as we want to integrate with as many ecosystem chains as possible. This will verify; Land Title Deed (Owners), Planning Code, Authority & Planning Status (once we start using the Rococo testnet). We will send you a link to this pallet once the latest update has been merged… its just we are experiencing some dependencies conflicts… Acurast are resolving these hopefully later today.

I hope we have delivered everything to your satisfaction.

As you will appreciate this is quite a large and complex project, which we are totally committed to. As a result, especially in the short term until we secure the seed funding, the W3F grant will be a life-line to help to realise this project and bring much needed users to the entire ecosystem.

Best Regards,

Richard

dsm-w3f commented:
@xcavate2022 thank you for the improvements and fixes. I did a new round of evaluation but some problems remain. Please see the details in the [evaluation document](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md). Let me know when I can continue with this evaluation.

RecrafterH commented:
@dsm-w3f sorry for didn't clarify this. You can build the contract by cd into the contracts/loan folder and call "cargo contract build --release". This works fine.

xcavate2022 commented:
﻿Hi Diogo,The React frontend is not completed yet (this was never a deliverable)… The instructions are to demonstrate that is how it will look when completed (these are based on the Figma designs). The W3F grant application deliverable was based only on having a simple “frontend” / Polkadot.js to demonstrate the ability to create a real estate loan using ink!… this was approved by Keegan (see his earlier ***@***.*** thanks for the update!To answer your question, the evaluator will want to see that the core software works as expected according to the milestone table(s). Usually the easiest way is typically to provide a Dockerfile or Docker Compose to spin up any nodes, unit tests and/or integration tests, e2e tests, etc. and a testing guide with a demo/tutorial of any working software. Essentially, the delivery should be broken down into the same deliverables in the original application.After taking a look at your application, I see that your front-end focus was rather small, with one deliverable noting that there will be a "working webpage". So I think in your case a static web page app would suffice.That being said, it's entirely possible for us to use and test your contracts and pallets using only Polkadot.js and not a custom front-end.We are a small team that are dependent on this grant to carry on with this build. Can you please indicate what deliverables are still outstanding.Your support is appreciated in advance.Best Regards,RichardSent from my iPhoneOn 18 Sep 2023, at 20:21, Diogo ***@***.***> wrote:﻿
@xcavate2022 thank you for the improvements and fixes. I did a new round of evaluation but some problems remain. Please see the details in the evaluation document. Let me know when I can continue with this evaluation.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

xcavate2022 commented:
Hi Diogo,Many thanks for your V3 evaluation… I will speak to my team and respond once reviewed.Best Regards,RichardSent from my iPhoneOn 18 Sep 2023, at 20:21, Diogo ***@***.***> wrote:﻿
@xcavate2022 thank you for the improvements and fixes. I did a new round of evaluation but some problems remain. Please see the details in the evaluation document. Let me know when I can continue with this evaluation.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

xcavate2022 commented:
Hi Diogo,

Please see our responses below in red:
Evaluation V3 (responses)

 <https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md#node>Node

I tried again to build using the newly indicated repository <https://github.com/XcavateBlockchain/Xcavate_Node> with Docker to check if I could build with success but I received the same error.

As previously stated although we can create a Docker image we cannot interact with the Frontend and Backend, so to save any confusion we have deleted the Docker file from the repo… please try to build the node using VS code

 <https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md#contract>Contract

I tried to build the contract in the new contract repository <https://github.com/XcavateBlockchain/xcavate_loan_contract.git> but I received this error.

Please try to CD in to the folder as per the read.me <http://read.me/> file

Upload the contract in to polkadot.js as per the testing video

 <https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md#decentralized-developer-loan-dapp>Decentralized Developer Loan dApp

This time, I was able to connect my wallet using the latest test version of Sporran. When I tried to link my credentials to my DID, I received this error in the backend.

Please see revised https://xcavate.io/XcavateDeveloperLoanUserInstructions.pdf

I updated the .env with the AWS and Crust network info provided in this comment <https://github.com/w3f/Grant-Milestone-Delivery/pull/992#issuecomment-1718973383>, and this solved the problem. I would like to know what is being stored in the AWS. Is this info necessary for the application? If It is, it needs to be in the .env.example and in the documentation with some instructions on how to get these variables.

There are only images and text being stored in AWS… just for testing purposes… The enhanced MVP will only use Crust (based on DID credential self sovereignty, real estate project application attachments / uploads & NFT json file data)

When I tried to Request a loan, I received 400 Bad Request in the console and this error in the backend when trying to submit in the last window.

The frontend is functioning now to receive the real estate developer loan application data in to Mongo DB - but for the full functionality we ask you to test via polkadot.js (as per the testing video)

The request info:

This Real Estate Developer Loan application requested information will be verified by an Acurast Oracle… however their API is showing dependency issues… we have reported the problem with Acurast have been informed an update is scheduled to be released by them soon.

I have some minor problems and questions. Should I be able to create a new DID using the front end? This step is on the PDF (pages 3, 4, and 5), but I didn't saw it.

The DID creation is based on using the Kilt Protocol… for testing purposes it uses the Peregrine Testnet
Please see revised https://xcavate.io/XcavateDeveloperLoanUserInstructions.pdf that show the correct procedure to create a DID and add & verify credentials

Some input boxes in the documentation are select boxes, but when I was testing, all the input boxes were input text.

When I uploaded the image, the "Click to upload or drag and drop" box didn't change it, so I tried a few times to upload the image but I gave up and submitted it. I checked the console and the PNG was sent. I tested using Firefox. This don't look like the expected behavior.

Please see revised https://xcavate.io/XcavateDeveloperLoanUserInstructions.pdf

I know if not perfect but I was under the impression that the W3F grant is to help and nurture teams that are looking to enhance the ecosystem with new real world use cases… our GTM version will do this by leveraging a number of Parachains using XCM

Best Regards,

Richard

Richard J Houldsworth
m 07798 666 710
www.xcavate.io


CONFIDENTIALITY: This email and its attachments are confidential to the intended recipient only. They may not be used by, disclosed to or copied in any way to anyone other than the intended recipient. If this email is received in error, please contact ***@***.***
 
Disclaimer STRICTLY CONFIDENTIAL The contents of this email and any attachments are confidential and intended solely for the organisation or individual to whom it is addressed. If you are not the intended recipient, you must not copy, print, forward, distribute or disseminate the information, or take any action in reliance of it. If you have received this message in error, do not open any attachment but please notify the sender above and delete this email from your system.






> On 18 Sep 2023, at 20:21, Diogo ***@***.***> wrote:
> 
> 
> @xcavate2022 <https://github.com/xcavate2022> thank you for the improvements and fixes. I did a new round of evaluation but some problems remain. Please see the details in the evaluation document <https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md>. Let me know when I can continue with this evaluation.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/w3f/Grant-Milestone-Delivery/pull/992#issuecomment-1724233573>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/A2OSX7BGSJDSYYQPJ524YPLX3CNMXANCNFSM6AAAAAA4IZVZMA>.
> You are receiving this because you were mentioned.
>

dsm-w3f commented:
@xcavate2022 thank you for the answers and fixes. I still having some problems. Please see the [evaluation document](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/xcavate_1_dsm-w3f.md) for details. Let me know when I can continue this evaluation.

xcavate2022 commented:
Hi Diogo,

Apologies I know this must be frustrating for you… I think there has been some confusion about the issues and what is still outstanding (the error is to do with the .env file in the MVP_Backend because if we put the correct signature in the repo readme.md file AWS closes my account because they say its a security breach to be publicly available).

I’ve revised the PDF instructions (to include the correct .env file info) so hopefully they are easier to follow now plus there are more screenshots to make things clearer.

https://xcavate.io/XcavateDeveloperLoanUserInstructions.pdf <https://xcavate.io/XcavateDeveloperLoanUserInstructions.pdf>

We believe the Verification Pallet (which verifies the off chain data) is the only thing outstanding and as soon as Acurast update the dependencies (hopefully by the end of this week) that should work too (I think they are at Sub0 the same as you so I think it makes things a little trickier to get done).

Best Regards,

Richard

meanking commented:
Hi @dsm-w3f, thanks for your efforts to evaluate our deliverables.
I think you got the `uploading error` with the AWS configuration. So I suggest you use this content of `.env` instead of yours. And please let me know if you get the same error.

```
PORT="9090"
JWT_SECRET="XCavate JWT Secret"
URL=http://localhost:9090

BLOCKCHAIN_ENDPOINT=wss://peregrine.kilt.io/parachain-public-ws
DID=did:kilt:4skimcqA5SDHsp4K6XM6nQVZSuCPAixbjW6MUok6e5uJqtuf

SECRET_PAYER_MNEMONIC=forest turn anchor because angry miracle slot unhappy claim blood champion dolphin
SECRET_AUTHENTICATION_MNEMONIC=cage tunnel resist radio lab cost quick slight axis mad ankle era
SECRET_ASSERTION_METHOD_MNEMONIC=view science pistol skull enlist bleak wave category lawn real drill balcony
SECRET_KEY_AGREEMENT_MNEMONIC=curtain chest safe roast place avocado faculty duck dial bless pill mutual

ADMIN_USERNAME=example
ADMIN_PASSWORD=attester

# MongoDB
MONGO_DB_URL=mongodb+srv://almacosm:Securitypro1@cluster0.mchtlqy.mongodb.net/xcavate-dev

# AWS S3
S3_ACCESS_KEY=AKIA3CXDWBJJJHPJBUBJ
S3_SECRET_KEY=IoevHCuSBy2jmmyIjSAwWcOaAArZEqeGF7K5cdDb
AWS_REGION=us-east-1
S3_BUCKET=xcavate-cdn
CDN_URL=https://xcavate-cdn.s3.amazonaws.com

# Crust network admin
CRUST_NETWORK_ADMIN_MNEMONIC=shoot coil invest fever output approve follow eight parent soul muffin few
CRUST_NETWORK_ADMIN_ADDRESS=5D7hbnc9oH5Dfhv1txZqyCR2vroHbXGKgk8S2jt1L3jZAieK
CRUST_NETWORK_ENDPOINT=wss://rpc-rocky.crust.network
CRUST_IPFS_W3GW=https://crustipfs.xyz
```

I am looking forward to hearing from you soon for the result with this.

xcavate2022 commented:
﻿﻿Hi Diogo,Many thanks for your recent assessment.Unfortunately Acurast have just commented that they cannot deliver the dependency updates…“It will not be released this week. The new substrate version has many breaking changes and was not our main focus this week. I will ping you and @Xcavate_2022 once we have a staging version with the changes.”So my question is… If we removed that deliverable and the frontend deliverable… how much would we need to reduce the grant amount down to? (Just a guide please so my team and I can make an informed decision). Also how long would this process take?If we changed the Docker deliverable to VS Code instructions and provided those, would that be sufficient to satisfy that deliverable?Best Regards,RichardSent from my iPhoneOn 21 Sep 2023, at 20:07, Diogo ***@***.***> wrote:﻿
@xcavate2022 thank you for the improvements and fixes. I still have some problems with the application but it could be related to the Acurast problem. The documentation still needs improvements. Please see the details in the evaluation document. Let me know when I can continue this evaluation.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

dsm-w3f commented:
@xcavate2022 thank you for the answer. How much discount you will give is up to you. We expect that to be proportional to the effort spent in the development of the deliverables that would be removed. The process is quite fast and consists of opening a PR changing the application document and explaining in the description of the PR the reason for the amendment. If you want to wait for Acurast to solve the problem is okay as well. Just let me know what you plan to do, and also when the docs are updated to check it again.

xcavate2022 commented:
Hi Diogo,

We have submitted a PR to remove the two deliverables (Verification Pallet & Frontend)

We have also tweaked the DAO deliverable & the Docker

Hopefully everything is now clearer with the creation of these documents


https://xcavate.gitbook.io/xcavate-documentation/￼
Introduction
xcavate.gitbook.io


I look forward to your feedback.

Best Regards,

Richard

> On 22 Sep 2023, at 13:07, Diogo ***@***.***> wrote:
> 
> 
> @xcavate2022 <https://github.com/xcavate2022> thank you for the answer. How much discount you will give is up to you. We expect that to be proportional to the effort spent in the development of the deliverables that would be removed. The process is quite fast and consists of opening a PR changing the application document and explaining in the description of the PR the reason for the amendment. If you want to wait for Acurast to solve the problem is okay as well. Just let me know what you plan to do, and also when the docs are updated to check it again.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/w3f/Grant-Milestone-Delivery/pull/992#issuecomment-1731307334>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/A2OSX7AWUEAZO4CJ7VIZF5DX3V5QFANCNFSM6AAAAAA4IZVZMA>.
> You are receiving this because you were mentioned.
>

dsm-w3f commented:
@xcavate2022 after the [amendment of the grant application](https://github.com/w3f/Grants-Program/pull/1995), the remainder deliverables are working well. The milestone is accepted. Please submit the invoice reflecting the new price for the milestone. I'll forward it internally and the payment should take place within two weeks.

xcavate2022 commented:
Thats fantastic - many thanks for all your help :)

Please see amended invoice attached

Best Regards,

Richard

xcavate2022 commented:
So sorry but I cannot find the password / seed phrase for the wallet on the previous invoice. Can you use this one please?

Best Regards,

Richard

xcavate2022 commented:
Hi All,

I hope you are well?

I don’t suppose someone can clarify when the Xcavate milestone 1 grant will be paid out? (sorry but a few of my team are asking me)

Also I have submitted this new PR for milestone 2 https://github.com/XcavateBlockchain/Grants-Program/blob/master/applications/Xcavate.md <https://github.com/XcavateBlockchain/Grants-Program/blob/master/applications/Xcavate.md>

Can you clarify if there is anything else I need to do before this is reviewed?

Best Regards,

Richard

dsm-w3f commented:
@xcavate2022 As you made an amendment to the application we need a new invoice with the adjusted value to be able to pay your team. Please send it using the same form that you used before for sending the invoice. After that, the payment processing time could take up to two weeks. Let me know when you submit the invoice to forward it internally.

xcavate2022 commented:
Thanks - have just submitted the revised invoice

semuelle commented:
Hi @xcavate2022. The payment address in the invoice does not match the one in the application. They need to be the same, so you have to either submit an invoice with the address from the application or submit an amendment PR to the contract with the address from the invoice.

xcavate2022 commented:
Sorry about that - had issues with that wallet but those have been resolved…. wallet address changed on invoice to reflect the same as the application

RouvenP commented:
hi @xcavate2022 we transferred the payment today
